### PR TITLE
TINY-6625: Table plugin tries to perform operations on tables that contain the (inline) editor

### PR DESCRIPTION
**Pull request by vpyshnenko**
Related Ticket: TINY-6625

Description of Changes:
* Placeholder text

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)
* [x] License headers added on new files (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

**Review by metricjs — APPROVED** [review]
Looks good to me! I do wonder if we should add more test cases, e.g. checking that `mceTableDelete` doesn't work, and maybe that things like adding a column shouldn't work? I'm not sure what would be the best things to test though, and worried the scope might blow out a lot. If you haven't already, it might be good to at least manually test a few things like that and see if anything else breaks?

**Comment by vpyshnenko** [comment]
Good point. I've checked and turns out the outside table is still accessible through commands. I'm gonna fix it.

File: modules/tinymce/src/plugins/table/test/ts/browser/InlineEditorInsideTableTest.ts [hunk 1]
```diff
@@ -35,13 +39,97 @@ describe('browser.tinymce.plugins.table.InlineEditorInsideTableTest', () => {
     inline: true,
     base_url: '/project/tinymce/js/tinymce',
     plugins: 'table',
-    menubar: false,
+    menubar: 'table',
     statusbar: false
   }, setupElement, [ Plugin, Theme ]);
 
+  const focusEditor = (editor: Editor) => {
```
Comment (lnewson) [inline modules/tinymce/src/plugins/table/test/ts/browser/InlineEditorInsideTableTest.ts:46]: Why is this needed? Can we just use `editor.focus()` instead, as we generally don't like to use internal core modules like this?

Additionally, this seems to be called in every test, so can we move this to a `beforeEach` instead please?

- [ ] Use `editor.focus()` and consider a `beforeEach` hook instead please.

File: modules/tinymce/src/plugins/table/test/ts/browser/InlineEditorInsideTableTest.ts [hunk 2]
```diff
@@ -35,13 +39,97 @@ describe('browser.tinymce.plugins.table.InlineEditorInsideTableTest', () => {
...
+      const td = editor.getBody().parentElement;
+      assert.notEqual(td.style.backgroundColor, 'pink');
+    });
+
+    it('TINY-6625: mceTableInsertRowBefore', () => {
+      const editor = hook.editor();
+      focusEditor(editor);
+      const beforeCount = count('tr');
+      editor.execCommand('mceTableInsertRowBefore');
+      assert.equal(count('tr'), beforeCount);
```
Comment (lnewson) [inline modules/tinymce/src/plugins/table/test/ts/browser/InlineEditorInsideTableTest.ts:84]: For debugging purposes, it could be better if the count function returned the array of elements and then used `assert.lengthOf` as that means it can print what it found in the error message.

**Review by lnewson — APPROVED** [review]

**Review by lnewson — APPROVED** [review]

**Review by ltrouton — APPROVED** [review]
